be/src/storage/rowset/rowset_writer.cpp

```diff
@@ -210,9 +210,15 @@ Status RowsetWriter::flush_segment(const SegmentPB& segment_pb, butil::IOBuf& da
         // 2. flush segment file
         auto writer = std::make_unique<SegmentFileWriter>(wfile.get());
 
-        size_t remaining_bytes = segment_pb.data_size();
+        butil::IOBuf segment_data;
+        int64_t remaining_bytes = data.cutn(&segment_data, segment_pb.data_size());
```

Comment:
I have a question. If the segment_pb.data_size() is a approximate value, the un-equality may occur frequently?

Reply:
data_size not approximate value, cut_into_writer may write approximate size data.

Reply:
if `cut_into_writer` write size equal the IOBuf size , the write will be actually, otherwise not.